### PR TITLE
Dogukan/etabs connector poc

### PR DESCRIPTION
**Pull request by dogukankaratas**
First attempt to implement DUI3 to ETABS. :tada: 

etabsDUI3

### CODE REVIEW

**Comment by bjoernsteinhagen** [comment]
@dogukankaratas thinking about folder directories: `ETABS/22/Speckle.Connectors.ETABS22` ? For future versions (same approach as `Rhino`)

image

**Comment by bjoernsteinhagen** [comment]
Working on my side too, nice @dogukankaratas! 🚀 

image

**Review by bjoernsteinhagen — COMMENTED** [review]
Not to be sorted in this PR but something for us to keep in mind:

- Manual loading of the `.dll` file to get the plugin in ETABS? No way to get it recognised automatically? Maybe we contact ETABS support.

**Review by bjoernsteinhagen — APPROVED** [review]

**Comment by oguzhankoral** [comment]
@dogukankaratas @bjoernsteinhagen please do not merge this PR in before having approval from @JR-Morgan or @adamhathcock

**Comment by dogukankaratas** [comment]
**CI Build Failure:**
CI checks are failing due to a local DLL dependency for CSIAPIv1. We're currently using a local DLL because there is no official NuGet package for it.

**Background:**
- In the previous version we were using: https://www.nuget.org/packages/CSiAPIv1 which is net48 compatible. The latest CSI API has been updated for net8 compatibility

**Required Action:**
- Need to publish a new NuGet package for the .NET 8 compatible version
- @JR-Morgan is already aware of this situation

**Note:** Once published, we can replace the local DLL reference with the new NuGet package.

**Comment by bjoernsteinhagen** [comment]
I suspect `Speckle.Connectors.sln` and `Local.sln` are out of sync following all the folder movement and renaming from `ETABS/Speckle.Connector.ETABSS22.csproj` to `CSI/ETABS22/Speckle.Connectors.ETABS22.csproj`, @JR-Morgan able to give me a hand with this tomorrow?

But on a good note, we're interacting with the model and selecting things.

image

File: Connectors/CSi/Speckle.Connectors.CSiShared/Properties/launchSettings.json
Comment (oguzhankoral) [inline Connectors/CSi/Speckle.Connectors.CSiShared/Properties/launchSettings.json:1]: wondering why ETABS 22`launchSettings.json` is in shared project 🤔

File: Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// NOTE: Plugin entry point must match the assembly name, otherwise hits you with a "Not found" error when loading plugin
+// TODO: Move ETABS implementation to csproj as part of CNX-835 and/or CNX-828
+namespace Speckle.Connectors.ETABS22;
```
Comment (oguzhankoral) [inline Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs:3]: wrong namespace?

**Review by oguzhankoral — CHANGES_REQUESTED** [review]

File: Connectors/CSi/Speckle.Connectors.CSiShared/Properties/launchSettings.json
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Properties/launchSettings.json:1]: Copy paste / error. Will remove :)

File: Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs (hunk 1 above)
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs:3]: Temp solution, that's why I marked as TODO. ETABS expects cPlugin and Form1 namespace match assembly name. When it was under CSiShared, it gave the "Not Found" error. Will make it more elegant in next series of Ticket(s)

File: Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Speckle.Connectors.Common;
+using Speckle.Connectors.CSiShared;
+using Speckle.Connectors.CSiShared.HostApp;
+using Speckle.Connectors.DUI.WebView;
+using Speckle.Sdk.Host;
+
+// NOTE: Plugin entry point must match the assembly name, otherwise hits you with a "Not found" error when loading plugin
+// TODO: Move ETABS implementation to csproj as part of CNX-835 and/or CNX-828
+namespace Speckle.Connectors.ETABS22;
```
[the remark this answers is not in the document]
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs:11]: Same as cPlugin.cs :)

[the remark this answers is not in the document]
Comment (clairekuang) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs:11]: Could we also rename this form to "SpeckleForm" or something more descriptive

[the remark this answers is not in the document]
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs:11]: This caused us headache 😆 It needs to (for some reason) be named `Form1.cs` @dogukankaratas and I can't explain yet.

File: Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs (hunk 1 above)
Comment (oguzhankoral) [reply Connectors/CSi/Speckle.Connectors.CSiShared/cPlugin.cs:3]: very annoying

**Review by oguzhankoral — APPROVED** [review]

File: Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamhathcock) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs:11]: this is gross

File: Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  public void SetSapModel(ref cSapModel sapModel, ref cPluginCallback pluginCallback)
+  {
+    _sapModel = sapModel;
+    _pluginCallback = pluginCallback;
+
+    // NOTE: Update the form to initialize the CSiSharedApplicationService when we receive "sapModel"
+    // Ensures service ready to use by other components
+    var csiService = Container!.GetRequiredService<ICSiApplicationService>();
```
Comment (adamhathcock) [inline Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs:43]: don't use the `!` operator

File: Connectors/CSi/Speckle.Connectors.CSiShared/HostApp/CSiSharedDocumentModelStore.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using Speckle.Connectors.DUI.Utils;
+
+namespace Speckle.Connectors.CSiShared.HostApp;
+
+public class CSiSharedDocumentModelStore : DocumentModelStore
+{
+  public CSiSharedDocumentModelStore(IJsonSerializer jsonSerializerSettings)
+    : base(jsonSerializerSettings) { }
+
+  protected override void HostAppSaveState(string modelCardState) => throw new NotImplementedException();
```
Comment (adamhathcock) [inline Connectors/CSi/Speckle.Connectors.CSiShared/HostApp/CSiSharedDocumentModelStore.cs:11]: does this not throw yet?

**Review by adamhathcock — CHANGES_REQUESTED** [review]

File: Connectors/CSi/Speckle.Connectors.CSiShared/HostApp/CSiSharedDocumentModelStore.cs (hunk 4 above)
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/HostApp/CSiSharedDocumentModelStore.cs:11]: Nope, not yet

File: Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs (hunk 3 above)
Comment (bjoernsteinhagen) [reply Connectors/CSi/Speckle.Connectors.CSiShared/Form1.cs]: Done

**Review by adamhathcock — APPROVED** [review]

**Pull request merged**